nixos/modules/services/networking/syncthing.nix

```diff
@@ -10,6 +10,28 @@ let
   settingsFormat = pkgs.formats.json { };
   cleanedConfig = converge (filterAttrsRecursive (_: v: v != null && v != {})) cfg.settings;
 
+  # syncthing supports serving the gui over unix sockets. if that happens, the
+  # api is serverd over the unix socket as well. this module uses the api to
+  # merge the configuration. curl is used for this and supports unix sockets.
+  # however the arguments are slightly different from an http request. the
+  # function `curlAddressArgs` returns a given path the correct string to make
+  # curl connect over either http(s) or unix. string which tells curl how to
+  # connect to the address. this works for https and unix.
+  curlAddressArgs = path: let
+    matchResult = builtins.match "^unix://(.*)" cfg.guiAddress;
+  in
+    if matchResult == null
+    # for http and https we do nothing
+    then "${cfg.guiAddress}/${path}"
+    # for unix sockets use the `--unix-socket` flag to tell curl
+    # where the socket is located in the filesystem. curl still
+    # expects a host segment in the http target location. that means
+    # the location must not start with a `/` character. that is why
+    # there is the `.` character at the end of the string. the dot
+    # will not show up in the actual http request.
+    else "--unix-socket ${builtins.head matchResult} .${path}"
```

Comment:
We could be a little more explicit here and do:
```
else "--unix-socket ${builtins.head matchResult} http://.${path}"
```

Reply:
oh that makes sense!